addons/mrp/models/product.py

```diff
@@ -195,3 +195,18 @@ def action_open_quants(self):
             res['context']['single_product'] = False
             res['context'].pop('default_product_tmpl_id', None)
         return res
+
+    def _search_qty_available_new(self, operator, value, lot_id=False, owner_id=False, package_id=False):
+        '''extending the method in stock.product to take into account kits'''
+        product_ids = set(super(ProductProduct, self)._search_qty_available_new(operator, value, lot_id, owner_id, package_id))
+        boms = self.env['mrp.bom'].read_group([('type', "=", 'phantom')], ['product_tmpl_id', 'ids:array_agg(id)'], ['product_tmpl_id', 'id'])
```

Comment:
this is grouping by product_tmp_id, but then using only the first bom found.

I think it should be checked for each bom, so something like:

boms = self.env['mrp.bom'].search([('type', "=", 'phantom')]).read([('product_tmpl_id')], load=False) # result like [{'id': 3, 'product_tmpl_id': 5}]